tools/flang2/flang2exe/main.cpp

```diff
@@ -428,9 +429,18 @@ main(int argc, char *argv[])
     if (!process_input(argv[0], &need_constructor))
       break;
 
+    // We cannot reuse gbl.func_count here since gbl.func_count must be
+    // incremented before calling process_input().
+    ilm_units_seen++;
+
   } while (!gbl.eof_flag);
 
-  cg_llvm_end();
+  if (ilm_units_seen == 0) {
+    errwarn((error_code_t)6);
+  }
+
+  if (FTN_HAS_INIT())
```

Comment:
Nit: A comment here on what FTN_HAS_INIT is and when it is set might be useful.

Reply:
Done.